tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs

```diff
@@ -307,6 +309,225 @@ where
 
     executor.setup(&cks, sks);
 
+    for _ in 0..NB_TESTS {
+        let clear = rng.gen::<u64>() % modulus;
+        let clear_shift = rng.gen::<u32>();
+
+        let ct = cks.encrypt(clear);
+
+        // case when 0 <= rotate < nb_bits
+        {
+            let clear_rotate = clear_shift % nb_bits;
+            let shift = cks.encrypt(clear_rotate as u64);
+            let encrypted_result = executor.execute((&ct, &shift));
+            let decrypted_result: u64 = cks.decrypt(&encrypted_result);
+            let expected = rotate_left_helper(clear, clear_rotate, nb_bits);
+            assert_eq!(expected, decrypted_result);
+        }
+
+        // case when shift >= nb_bits
+        {
+            let clear_rotate = clear_shift.saturating_add(nb_bits);
+            let shift = cks.encrypt(clear_rotate as u64);
+            let encrypted_result = executor.execute((&ct, &shift));
+            let decrypted_result: u64 = cks.decrypt(&encrypted_result);
+            // When nb_bits is not a power of two
+            // then the behaviour is not the same
+            let true_nb_bits = nb_bits;
+            let mut nb_bits = nb_bits;
+            if !nb_bits.is_power_of_two() {
+                nb_bits = nb_bits.next_power_of_two();
+            }
+            let expected = rotate_left_helper(clear, clear_rotate % nb_bits, true_nb_bits);
+            assert_eq!(expected, decrypted_result);
+        }
+    }
+}
+
+pub(crate) fn unchecked_rotate_right_test<P, T>(param: P, mut executor: T)
```

Comment:
I haven't moved the content of `tests_cases_unsigned` around yet so ok to keep your code here.